contracts/container/contract.go

```diff
@@ -491,6 +494,143 @@ func List(owner []byte) [][]byte {
+	}
+
+	ctx := storage.GetContext()
+	multiaddr := common.AlphabetAddress()
+	common.CheckAlphabetWitness(multiaddr)
+
+	commonPrefix := append([]byte{nextEpochNodesPrefix}, cID...)
+	commonPrefix = append(commonPrefix, byte(placementVector))
+
+	counter := 0
+	c := storage.Find(ctx, commonPrefix, storage.KeysOnly|storage.Backwards)
+	if iterator.Next(c) {
+		counterRaw := iterator.Value(c).([]byte)[1+interop.Hash256Len:]
+		counter = counterFromBE(counterRaw)
+	}
+
+	for _, publicKey := range publicKeys {
+		if len(publicKey) != interop.PublicKeyCompressedLen {
+			panic(cst.ErrorInvalidPublicKey + ": length: " + std.Itoa10(len(publicKey)))
+		}
+
+		counter++
+
+		storageKey := append(commonPrefix, counterBE(counter)...)
+		storage.Put(ctx, storageKey, publicKey)
+	}
+}
+
+func counterBE(c int) []byte {
+	rawCounter := std.Serialize(c)
+	res := []byte{rawCounter[0], rawCounter[1]} // first is type, second is length
+	for i := len(rawCounter) - 1; i > 1; i-- {  // LE to BE
+		res = append(res, rawCounter[i])
+	}
+
+	return res
+}
+
+func counterFromBE(b []byte) int {
+	res := []byte{b[0], b[1]}         // first is type, second is length
+	for i := len(b) - 1; i > 1; i-- { // BE to LE
+		res = append(res, b[i])
+	}
+
+	return std.Deserialize(res).(int)
+}
+
+// CommitContainerListUpdate commits container list changes made by
+// [AddNextEpochNodes] calls in advance. Replicas must correspond to
+// ordered placement policy (REP clauses). If no [AddNextEpochNodes]
+// have been made, it clears container list. Makes "ContainerUpdate"
+// notification with container ID after successful list change.
+// Call must be signed by the Alphabet nodes.
+func CommitContainerListUpdate(cID interop.Hash256, replicas []uint8) {
+	if len(cID) != interop.Hash256Len {
+		panic(cst.ErrorInvalidContainerID + ": length: " + std.Itoa10(len(cID)))
+	}
+
+	ctx := storage.GetContext()
+	multiaddr := common.AlphabetAddress()
+	common.CheckAlphabetWitness(multiaddr)
+
+	oldNodesPrefix := append([]byte{nodesPrefix}, cID...)
+	newNodesPrefix := append([]byte{nextEpochNodesPrefix}, cID...)
+	replicasPrefix := append([]byte{replicasNumberPrefix}, cID...)
+
+	oldNodes := storage.Find(ctx, oldNodesPrefix, storage.KeysOnly)
+	for iterator.Next(oldNodes) {
+		oldNode := iterator.Value(oldNodes).(string)
+		storage.Delete(ctx, oldNode)
+	}
+
+	newNodes := storage.Find(ctx, newNodesPrefix, storage.None)
+	for iterator.Next(newNodes) {
+		newNode := iterator.Value(newNodes).(struct {
+			key []byte
+			val []byte
+		})
+
+		storage.Delete(ctx, newNode.key)
+
+		newKey := append([]byte{nodesPrefix}, newNode.key[1:]...)
+		storage.Put(ctx, newKey, newNode.val)
+	}
+
+	rr := storage.Find(ctx, replicasPrefix, storage.KeysOnly)
+	for iterator.Next(rr) {
+		oldReplicasNumber := iterator.Value(rr).([]byte)
+		storage.Delete(ctx, oldReplicasNumber)
+	}
+
+	// nolint:gosimple // this is contract, ranging over nil leads to conversion error
```

Comment:
File an issue for compiler then.

Reply:
https://github.com/nspcc-dev/neo-go/issues/3608